File: classwork/im-v/primes/primes.py
#!/usr/bin/env python3

import sys

def is_prime(x):
    if x < 2:
        return False
    for i in range(2, int(x**0.5)+1):
        if x % i == 0:
            return False
    return True

def main():
    if len(sys.argv) != 2:
        print("Usage: python3 primes.py <n>")
        sys.exit(1)

    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Please provide a valid integer.")
        sys.exit(1)

    primes = [i for i in range(1, n+1) if is_prime(i)]
    print("Primes up to", n, ":", primes)

if __name__ == "__main__":
    main()

File: classwork/im-v/primes/primes.js
#!/usr/bin/env node

function isPrime(x) {
  if (x < 2) return false;
  for (let i = 2; i <= Math.sqrt(x); i++) {
    if (x % i === 0) return false;
  }
  return true;
}

function main() {
  const args = process.argv.slice(2);
  if (args.length !== 1) {
    console.error("Usage: node primes.js <n>");
    process.exit(1);
  }

  const n = parseInt(args[0], 10);
  if (isNaN(n)) {
    console.error("Please provide a valid integer.");
    process.exit(1);
  }

  const primes = [];
  for (let i = 1; i <= n; i++) {
    if (isPrime(i)) primes.push(i);
  }

  console.log(`Primes up to ${n}:`, primes);
}

main();